File: parallel_execution/sources/loop.2.c
/* 
* @@name:	loop.2
* @@type:	C
* @@operation:	run
* @@expect:	success
* @@version:	omp_5.0
*/
#include  <stdio.h>
#define N 1024

int x[N][N];
int y[N], z[N]; 

void foo() {
// i-loop distributed across encountering league of teams
#pragma omp loop bind(teams)
    for (int i = 0; i < N; i++) {
    // this loop has an implicit bind(thread)
    #pragma omp loop
      for (int j = 0; j < N; j++) {
        x[i][j] += y[i]*z[i];
      }
    }
}

int main(){
 int error = 0;

  for (int i = 0; i < N; i++) {
    for (int j = 0; j < N; j++) {
       x[i][j] = 0;
    }
  }

  for (int i = 0; i < N; i++) {
    y[i] = i;
    z[i] = i+1;
  }

#pragma omp teams num_teams(4) 
  {
      foo();
  }

//check values
   for (int i = 0; i < N; i++) {
    for (int j = 0; j < N; j++) {
       if( x[i][j] != i * (i+1))
         error++;
    }
  }
  if(error) {
    printf("FAILED\n");
    return 1;
  }    
  printf("PASSED\n");   
  return 0;
} 

File: parallel_execution/sources/loop.2.f90
! @@name:	loop.2
! @@type:	F-free
! @@operation:	run
! @@expect:	success
! @@version:	omp_5.0
module xyz_data
  integer, parameter :: N=1024
  integer :: x(N,N)
  integer :: y(N), z(N)

contains 
  subroutine foo() 
    integer :: i, j

    !! i-loop distributed across encountering league of
    !! teams
    !$omp loop bind(teams)
    do i = 1, N
      !! this loop has an implicit bind(thread)
      !$omp loop
      do j = 1, N
        x(j,i) = x(j,i) + y(i)*z(i)
      end do
    end do
  end subroutine
end module

program main
 use xyz_data
 integer ::  error = 0

  do i = 1, N
    do j = 1, N
      x(j,i) = 0
    end do
  end do

  do i = 1, N
    y(i) = i
    z(i) = i + 1
  end do

  !$omp teams num_teams(4) 
  call foo()
  !$omp end teams

!!check values
  do i = 1, N
    do j = 1, N
      if( x(j,i) /= i * (i+1) ) then
        error = error + 1
      endif
    enddo
  enddo

  if(error .gt. 0) then
    print*, "FAILED"
    stop 1
  end if
  
  print*, "PASSED"

end program
